### ISSUE TITLE
good accuracy but bad prediction

### ISSUE THREAD

**Issue by rena1234567** [keras-team/keras#9939]
Hello everyone, I need help with this please, I using keras classifier the problem is that I got high accuracy but very bad prediction with the same data , the prediction classes should be 0,1,2,3,3,4,0. but I got all zeros, here is my code

from keras.models import Sequential
from keras.layers import Dense
from keras import optimizers
model = Sequential()
model.add(Dense(units=14, activation='relu', input_shape=(14,)))
model.add(Dropout(0.5))
model.add(Dense(units=14, activation='relu'))
model.add(Dropout(0.5))
model.add(Dense(units=14, activation='relu'))
model.add(Dense(units=5, activation='sigmoid'))
#model.add(Dense(units=5, activation='relu'))
#monitor = EarlyStopping(monitor='val_loss', min_delta=1e-3, patience=5, verbose=0, mode='auto')
#checkpointer = ModelCheckpoint(filepath="best_weights.hdf5", verbose=0, save_best_only=True) # save best model  
sgd = optimizers.SGD(lr=0.01, decay=1e-6, momentum=0.001, nesterov=True)
model.compile(optimizer=sgd,
               loss='categorical_crossentropy',
               metrics=['accuracy'])

model.fit(x_vals_train,y_train,validation_data=(x_vals_test,y_test),verbose=0,epochs=10)
#Evaluate the accuracy of our trained model
score = model.evaluate(x_vals_test, y_test,
                       batch_size=32, verbose=1)
print('Test score:', score[0])
print('Test accuracy:', score[1])
27134/27134 [==============================] - 1s 29us/step
Test score: 0.10602876026708943
Test accuracy: 0.9448293653718581

the predition
testset = np.loadtxt('G:/project/test.pcap_z/all_data_amount_7.csv', delimiter=',')

xtest = testset[:,0:14]

#x_test1 = np.nan_to_num(normalize_cols(xtest))

y_pred = model.predict(x_test1)
y_pred =y_pred.astype(int)
y_pred

array([[0, 0, 0, 0, 0],
       [0, 0, 0, 0, 0],
       [0, 0, 0, 0, 0],
       [0, 0, 0, 0, 0],
       [0, 0, 0, 0, 0],
       [0, 0, 0, 0, 0],
       [0, 0, 0, 0, 0],
       [0, 0, 0, 0, 0]])
please ,any help will be really appreciated.

**Comment by gusdlf93** [comment]
1.model.add(Dense(units=5, activation='sigmoid'))
->model.add(Dense(units=5, activation='softmax'))
2.sgd = optimizers.SGD(lr=0.01, decay=1e-6, momentum=0.001, nesterov=True)
->sgd = optimizers.SGD(lr=0.01, momentum=0.9, nesterov=True)
3.Where is you're validation loss and accuracy?
validation is very important informations.
